src/Domain/JacobianDiagnostic.cpp

```diff
@@ -0,0 +1,105 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/JacobianDiagnostic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain {
+template <size_t Dim, typename Fr>
+void jacobian_diagnostic(
+    const gsl::not_null<
+        tnsr::i<DataVector, Dim, typename Frame::ElementLogical>*>
+        jacobian_diag,
+    const Jacobian<DataVector, Dim, typename Frame::ElementLogical, Fr>&
+        analytic_jacobian,
+    const TensorMetafunctions::prepend_spatial_index<
+        tnsr::I<DataVector, Dim, Fr>, Dim, UpLo::Lo,
+        typename Frame::ElementLogical>& numeric_jacobian_transpose) {
+  destructive_resize_components(jacobian_diag,
+                                analytic_jacobian.begin()->size());
+  const size_t number_of_points = analytic_jacobian.begin()->size();
+  auto analytic_jacobian_sum =
+      tnsr::i<DataVector, Dim, typename Frame::ElementLogical>{number_of_points,
+                                                               0.0};
+  auto numerical_jacobian_sum =
+      tnsr::i<DataVector, Dim, typename Frame::ElementLogical>{number_of_points,
+                                                               0.0};
+  constexpr double eps_to_avoid_div_by_zero = 1.e-20;
+
+  // i_hat = logical frame index
+  // i = mapped frame index
```

Comment:
You can do this with 1 `Scalar<DataVector>` as temporary storage and just use `abs` instead of RMS.